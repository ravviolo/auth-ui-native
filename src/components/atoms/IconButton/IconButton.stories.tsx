import { action } from '@storybook/addon-actions';
import { color, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { IconButton } from './IconButton';

storiesOf('IconButton', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <IconButton
      color={color('color', '#4267B2')}
      icon={text('icon', 'f')}
      testID={text('testID', 'icon-btn-test-id')}
      onPress={action('onPress')}
    />
  ));
