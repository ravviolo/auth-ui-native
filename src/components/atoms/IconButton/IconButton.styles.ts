import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
  btn: {
    alignItems: 'center',
    borderRadius: 100,
    borderWidth: 2,
    height: 50,
    justifyContent: 'center',
    padding: 8,
    width: 50,
  },
  icon: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  pressed: {
    backgroundColor: colors.textWhite,
  },
});
