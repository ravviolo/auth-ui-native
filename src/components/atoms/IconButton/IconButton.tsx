import { StyleSheet, Text, TouchableHighlight } from 'react-native';

interface Props {
  icon: string;
  color: string;
  testID: string;
  onPress: () => void;
}

export const IconButton = ({ onPress, color, icon, testID }: Props) => {
  return (
    <TouchableHighlight
      style={[styles.btn, { borderColor: color }]}
      testID={testID}
      onPress={onPress}
    >
      <Text style={[styles.icon, { color }]}>{icon}</Text>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  btn: {
    alignItems: 'center',
    borderRadius: 100,
    borderWidth: 2,
    height: 50,
    justifyContent: 'center',
    padding: 5,
    width: 50,
  },
  icon: {
    fontSize: 18,
    fontWeight: 'bold',
  },
});
