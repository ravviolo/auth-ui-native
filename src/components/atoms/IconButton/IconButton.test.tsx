import { render, screen, fireEvent } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { IconButton } from './IconButton';

import type { Props } from './IconButton';

describe('IconButton', () => {
  const mockOnPress = jest.fn<Props['onPress'], []>();

  const props: Props = {
    color: 'cyan',
    icon: 'I',
    testID: 'icon-btn-test-id',
    onPress: mockOnPress,
  };

  it('should render button with specified color and icon', () => {
    render(<IconButton {...props} />);

    const iconBtn = screen.getByTestId('icon-btn-test-id');

    expect(iconBtn).toHaveTextContent('I');
    expect(iconBtn).toHaveStyle({ borderColor: 'cyan' });
    expect(iconBtn.children[0]).toHaveStyle({ color: 'cyan' });
  });

  it('should handle events - run handler once', () => {
    render(<IconButton {...props} />);

    const iconBtn = screen.getByTestId('icon-btn-test-id');

    fireEvent.press(iconBtn);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<IconButton {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
