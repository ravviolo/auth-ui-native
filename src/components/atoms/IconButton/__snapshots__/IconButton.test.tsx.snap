// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "borderWidth": 2,
        "height": 50,
        "justifyContent": "center",
        "padding": 8,
        "width": 50,
      },
      Object {
        "borderColor": "cyan",
      },
    ]
  }
  testID="icon-btn-test-id"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        Object {
          "color": "cyan",
        },
      ]
    }
  >
    I
  </Text>
</View>
`;
